aerospike-core/src/cluster/mod.rs

```diff
@@ -493,16 +485,13 @@ impl Cluster {
     }
 
     pub async fn get_node(&self, partition: &Partition<'_>) -> Result<Arc<Node>> {
-        let partitions = self.partitions();
-        let partitions = partitions.read().await;
-
-        if let Some(node_array) = partitions.get(partition.namespace) {
-            if let Some(node) = node_array.get(partition.partition_id) {
-                return Ok(node.clone());
-            }
-        }
+        let partitions = self.partition_write_map.read().await;
 
-        self.get_random_node().await
+        partitions
```

Comment:
Reading/writing to a random partition seems like a terrible idea. This way we get good error messages if the state is wrong.